clustered/ehcache-client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehind.java

```diff
@@ -76,7 +80,11 @@ private void process(long hash) {
               try {
                 if (result != null) {
                   if (result != currentState.get(key) && !(operation instanceof PutOperation)) {
-                    cacheLoaderWriter.write(result.getKey(), result.getValue());
+                    try {
+                      cacheLoaderWriter.write(result.getKey(), result.getValue());
+                    } catch (Exception e) {
+                      keyCleanUpMethod.accept(key);
```

Comment:
I would be very uncomfortable making this change without lots of test coverage to back it up. You should be terrified. The clustered store code is pretty different to the unclustered, and I'm really not convinced this is safe. Worst case scenario here is probably some kind of deadlock...